flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSource.java

```diff
@@ -111,24 +112,17 @@
 		return kafkaSource;
 	}
 
-	@Override
-	public int getNumberOfFields() {
-		return fieldNames.length;
-	}
-
-	@Override
 	public String[] getFieldsNames() {
-		return fieldNames;
+		return TableSource$class.getFieldsNames(this);
```

Comment:
What about to make `TableSource` abstract class? So that it can fit in with Java and Scala without involving something hack.

In this way, the `StreamTableSource` and `BatchTableSource` should be abstract class too.